core/src/snapshot_packager_service.rs

```diff
@@ -197,11 +231,46 @@ impl SnapshotPackagerService {
         pending_snapshot_packages.lock().unwrap().pop()
     }
 
+    /// Returns any pending fastboot snapshot packages
+    fn get_pending_fastboot_snapshot_package(
+        pending_snapshot_packages: &Mutex<PendingSnapshotPackages>,
+    ) -> Option<SnapshotPackage> {
+        pending_snapshot_packages.lock().unwrap().pop_fastboot()
+    }
+
     /// Performs final operations before gracefully shutting down
-    fn teardown(state: &TeardownState, snapshot_config: &SnapshotConfig) {
-        info!("Flushing account storages...");
+    fn teardown(state: TeardownState, snapshot_config: &SnapshotConfig) {
         let start = Instant::now();
-        for storage in &state.snapshot_storages {
+
+        let TeardownState {
+            snapshot_slot,
+            snapshot_storages,
+            bank_snapshot_package,
+        } = state;
+
+        if let Some(bank_snapshot_package) = bank_snapshot_package {
+            info!("Serializing bank snapshot...");
+            let result = snapshot_utils::serialize_snapshot(
```

Comment:
Need the timer here.


```suggestion
            info!("Serializing bank snapshot...");
            let start = Instant::now();
            let result = snapshot_utils::serialize_snapshot(
```